expression/expression.go

```diff
@@ -77,6 +79,41 @@ func (triggerExpression TriggerExpression) Get(name string) (interface{}, error)
 	}
 }
 
+// Validate returns error if triggers of type moira.ExpressionTrigger are badly formatted otherwise nil
+func (triggerExpression *TriggerExpression) Validate() error {
```

Comment:
Now looked at the front hanging when 400 is returned from `api/trigger/check`, this is fixed if you wrap the returned error in `ErrrInvalidExpression`, I have locally raised and tested this, so I suggest the following changes:
```go
// Validate returns error if triggers of type moira.ExpressionTrigger are badly formatted otherwise nil
func (triggerExpression *TriggerExpression) Validate() error {
	if triggerExpression.TriggerType != moira.ExpressionTrigger {
		return nil
	}
	if triggerExpression.Expression == nil || *triggerExpression.Expression == "" {
		return ErrInvalidExpression{internalError: fmt.Errorf("trigger_type set to expression, but no expression provided")}
	}
	expression := *triggerExpression.Expression
	env := map[string]interface{}{
		"ok":         moira.StateOK,
		"error":      moira.StateERROR,
		"warn":       moira.StateWARN,
		"warning":    moira.StateWARN,
		"nodata":     moira.StateNODATA,
		"t1":         triggerExpression.MainTargetValue,
		"prev_state": triggerExpression.PreviousState,
	}
	if triggerExpression.WarnValue != nil {
		env["warn_value"] = *triggerExpression.WarnValue
	}
	if triggerExpression.ErrorValue != nil {
		env["error_value"] = *triggerExpression.ErrorValue
	}
	for k, v := range triggerExpression.AdditionalTargetsValues {
		env[k] = v
	}
	if _, err := expr.Compile(
		strings.ToLower(expression),
		expr.Optimize(true),
		expr.Env(env),
	); err != nil {
		return ErrInvalidExpression{internalError: err}
	}
	return nil
}
```

Reply:
What about errors like this one when checking for nil / empty Expression. Will that not affect the front as well? :
```		
return fmt.Errorf("trigger_type set to expression, but no expression provided")
```

Reply:
I just did it the way it was in `Evaluate`, there absolutely all errors are wrapped up in `ErrrInvalidExpression`

Reply:
alright, fix coming in hot ))